### PR TITLE
workflows: upload to Linuxbrew org

### PR DESCRIPTION
**Pull request by iMichka**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/linuxbrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/linuxbrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [ ] Have you included the output of `brew gist-logs <formula>` of the build failure if your PR fixes a build failure. Please quote the exact error message.

-----

### CODE REVIEW

**Comment by Bo98** [comment]
Need to make sure Linuxbrew org is feature flagged to publish public pakages.

File: .github/workflows/dispatch-build-bottle.yml [hunk 1]
```diff
@@ -77,7 +77,7 @@ jobs:
           HOMEBREW_GITHUB_PACKAGES_TOKEN: ${{secrets.HOMEBREW_CORE_GITHUB_PACKAGES_TOKEN}}
         run: |
           cd ~/bottles
-          brew pr-upload --verbose --keep-old --no-commit --root-url="https://ghcr.io/v2/Homebrew/linuxbrew-core"
+          brew pr-upload --verbose --keep-old --no-commit --root-url="https://ghcr.io/v2/Linuxbrew/linuxbrew-core"
```
Comment (sjackman) [inline .github/workflows/dispatch-build-bottle.yml:80]:
```suggestion
          brew pr-upload --verbose --keep-old --no-commit --root-url="https://ghcr.io/v2/linuxbrew/core"
```
Org needs to be lowercase.
We're dropping the `homebrew-` or `linuxbrew-` prefix from the tap name.
I prefer `docker://ghcr.io/linuxbrew/core` over `https://ghcr.io/v2/linuxbrew/core`. That may first require PR https://github.com/Homebrew/brew/pull/11037

File: .github/workflows/publish-commit-bottles.yml [hunk 2]
```diff
@@ -66,7 +66,7 @@ jobs:
           HOMEBREW_GITHUB_API_TOKEN: ${{secrets.HOMEBREW_GITHUB_API_TOKEN}}
           HOMEBREW_GITHUB_PACKAGES_USER: brewtestbot
           HOMEBREW_GITHUB_PACKAGES_TOKEN: ${{secrets.HOMEBREW_CORE_GITHUB_PACKAGES_TOKEN}}
-        run: brew pr-pull --debug --workflows=tests.yml --no-commit --verbose --root-url="https://ghcr.io/v2/Homebrew/linuxbrew-core" ${{github.event.inputs.args}} ${{github.event.inputs.pull_request}}
+        run: brew pr-pull --debug --workflows=tests.yml --no-commit --verbose --root-url="https://ghcr.io/v2/Linuxbrew/linuxbrew-core" ${{github.event.inputs.args}} ${{github.event.inputs.pull_request}}
```
Comment (sjackman) [inline .github/workflows/publish-commit-bottles.yml:69]: Ditto above.

File: .github/workflows/dispatch-build-bottle.yml (hunk 1 above)
Comment (Bo98) [reply .github/workflows/dispatch-build-bottle.yml:80]: Uppercase Homebrew seems to work over at homebrew-core?

**Comment by iMichka** [comment]
I made the org lowercase, and renamed the token to be `LINUXBREW_CORE_GITHUB_PACKAGES_TOKEN`, to let developers know that this token is coming from somewhere else.

**Comment by iMichka** [comment]
> Need to make sure Linuxbrew org is feature flagged to publish public pakages.

How do I do this?

**Comment by Bo98** [comment]
Ask someone at GitHub. @MikeMcQuaid will know what to do.

**Comment by MikeMcQuaid** [comment]
> Need to make sure Linuxbrew org is feature flagged to publish public pakages.

Done already!

File: .github/workflows/dispatch-build-bottle.yml (hunk 1 above)
Comment (sjackman) [reply .github/workflows/dispatch-build-bottle.yml]: https://github.com/Homebrew/brew/blob/6d0275ff574bc3958455f26e7561847d8195168c/Library/Homebrew/github_packages.rb#L98-L99

Homebrew/brew lower-cases it for us, but it'd be more clear to use the lower case letters. Docker explicitly disallows upper case letters in the image name. See https://docs.docker.com/engine/reference/commandline/tag/

**Comment by sjackman** [comment]
Looks good to me!

File: .github/workflows/publish-commit-bottles.yml [hunk 3]
```diff
@@ -61,6 +61,13 @@ jobs:
           HOMEBREW_BINTRAY_KEY: ${{secrets.HOMEBREW_BINTRAY_KEY}}
         run: brew pr-pull --bintray-org=linuxbrew --keep-old --debug ${{github.event.inputs.args}} ${{github.event.inputs.pull_request}}
 
+      - name: Pull and upload bottles to GitHub Packages
+        env:
+          HOMEBREW_GITHUB_API_TOKEN: ${{secrets.HOMEBREW_GITHUB_API_TOKEN}}
+          HOMEBREW_GITHUB_PACKAGES_USER: brewtestbot
+          HOMEBREW_GITHUB_PACKAGES_TOKEN: ${{secrets.HOMEBREW_CORE_GITHUB_PACKAGES_TOKEN}}
+        run: brew pr-pull --debug --workflows=tests.yml --no-commit --verbose --root-url="https://ghcr.io/v2/linuxbrew/core" ${{github.event.inputs.args}} ${{github.event.inputs.pull_request}}
```
Comment (MikeMcQuaid) [inline .github/workflows/publish-commit-bottles.yml:69]:
```suggestion
          HOMEBREW_GITHUB_PACKAGES_USER: linuxbrewtestbot
          HOMEBREW_GITHUB_PACKAGES_TOKEN: ${{secrets.HOMEBREW_CORE_GITHUB_PACKAGES_TOKEN}}
        run: brew pr-pull --debug --workflows=tests.yml --no-commit --verbose --root-url="https://ghcr.io/v2/Linuxbrew/homebrew-core" ${{github.event.inputs.args}} ${{github.event.inputs.pull_request}}
```

File: .github/workflows/dispatch-build-bottle.yml [hunk 4]
```diff
@@ -71,6 +71,14 @@ jobs:
             --verbose \
             --debug
 
+      - name: Upload bottles to GitHub Packages
+        env:
+          HOMEBREW_GITHUB_PACKAGES_USER: brewtestbot
+          HOMEBREW_GITHUB_PACKAGES_TOKEN: ${{secrets.HOMEBREW_CORE_GITHUB_PACKAGES_TOKEN}}
+        run: |
+          cd ~/bottles
+          brew pr-upload --verbose --keep-old --no-commit --root-url="https://ghcr.io/v2/linuxbrew/core"
```
Comment (MikeMcQuaid) [inline .github/workflows/dispatch-build-bottle.yml:80]:
```suggestion
          HOMEBREW_GITHUB_PACKAGES_USER: linuxbrewtestbot
          HOMEBREW_GITHUB_PACKAGES_TOKEN: ${{secrets.HOMEBREW_CORE_GITHUB_PACKAGES_TOKEN}}
        run: |
          cd ~/bottles
          brew pr-upload --verbose --keep-old --no-commit --root-url="https://ghcr.io/v2/Linuxbrew/homebrew-core"
```

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Have updated this to use @LinuxbrewTestBot credentials and made this more consistent with Homebrew/homebrew-core (for now at least).

File: .github/workflows/dispatch-build-bottle.yml [hunk 5]
```diff
@@ -71,6 +71,14 @@ jobs:
             --verbose \
             --debug
 
+      - name: Upload bottles to GitHub Packages
+        env:
+          HOMEBREW_GITHUB_PACKAGES_USER: linuxbrewtestbot
+          HOMEBREW_GITHUB_PACKAGES_TOKEN: ${{secrets.HOMEBREW_CORE_GITHUB_PACKAGES_TOKEN}}
+        run: |
+          cd ~/bottles
+          brew pr-upload --verbose --keep-old --no-commit --root-url="https://ghcr.io/v2/Linuxbrew/homebrew-core"
```
Comment (MikeMcQuaid) [inline .github/workflows/dispatch-build-bottle.yml:80]:
```suggestion
          brew pr-upload --verbose --keep-old --no-commit --root-url="https://ghcr.io/v2/linuxbrew/core"
```

File: .github/workflows/publish-commit-bottles.yml [hunk 6]
```diff
@@ -61,6 +61,13 @@ jobs:
           HOMEBREW_BINTRAY_KEY: ${{secrets.HOMEBREW_BINTRAY_KEY}}
         run: brew pr-pull --bintray-org=linuxbrew --keep-old --debug ${{github.event.inputs.args}} ${{github.event.inputs.pull_request}}
 
+      - name: Pull and upload bottles to GitHub Packages
+        env:
+          HOMEBREW_GITHUB_API_TOKEN: ${{secrets.HOMEBREW_GITHUB_API_TOKEN}}
+          HOMEBREW_GITHUB_PACKAGES_USER: linuxbrewtestbot
+          HOMEBREW_GITHUB_PACKAGES_TOKEN: ${{secrets.HOMEBREW_CORE_GITHUB_PACKAGES_TOKEN}}
+        run: brew pr-pull --debug --workflows=tests.yml --no-commit --verbose --root-url="https://ghcr.io/v2/Linuxbrew/homebrew-core" ${{github.event.inputs.args}} ${{github.event.inputs.pull_request}}
```
Comment (MikeMcQuaid) [inline .github/workflows/publish-commit-bottles.yml:69]:
```suggestion
        run: brew pr-pull --debug --workflows=tests.yml --no-commit --verbose --root-url="https://ghcr.io/v2/linuxbrew/core" ${{github.event.inputs.args}} ${{github.event.inputs.pull_request}}
```

**Review by MikeMcQuaid — APPROVED** [review]
Actually, changed my mind, submitting PR to homebrew-core in https://github.com/Homebrew/homebrew-core/pull/74752

**Pull request merged**